tiledb/sm/serialization/array_schema.cc

```diff
@@ -117,35 +117,41 @@ Status filter_pipeline_to_capnp(
   return Status::Ok();
 }
 
-Status filter_pipeline_from_capnp(
-    const capnp::FilterPipeline::Reader& filter_pipeline_reader,
-    tdb_unique_ptr<FilterPipeline>* filter_pipeline) {
-  filter_pipeline->reset(tdb_new(FilterPipeline));
+tuple<Status, optional<shared_ptr<FilterPipeline>>> filter_pipeline_from_capnp(
+    const capnp::FilterPipeline::Reader& filter_pipeline_reader) {
   if (!filter_pipeline_reader.hasFilters())
-    return Status::Ok();
+    return {Status::Ok(), make_shared<FilterPipeline>(HERE())};
 
+  std::vector<Filter*> filter_list;
   auto filter_list_reader = filter_pipeline_reader.getFilters();
   for (auto filter_reader : filter_list_reader) {
     FilterType type = FilterType::FILTER_NONE;
-    RETURN_NOT_OK(filter_type_enum(filter_reader.getType().cStr(), &type));
-    tdb_unique_ptr<Filter> filter(FilterCreate::make(type));
+    RETURN_NOT_OK_TUPLE(
+        filter_type_enum(filter_reader.getType().cStr(), &type), nullopt);
+
+    Filter* filter(FilterCreate::make(type));
     if (filter == nullptr)
-      return LOG_STATUS(Status_SerializationError(
-          "Error deserializing filter pipeline; failed to create filter."));
+      return {
+          LOG_STATUS(Status_SerializationError(
+              "Error deserializing filter pipeline; failed to create filter.")),
+          nullopt};
 
     switch (filter->type()) {
       case FilterType::FILTER_BIT_WIDTH_REDUCTION: {
         auto data = filter_reader.getData();
         uint32_t window = data.getUint32();
-        RETURN_NOT_OK(
-            filter->set_option(FilterOption::BIT_WIDTH_MAX_WINDOW, &window));
+        RETURN_NOT_OK_TUPLE(
+            filter->set_option(FilterOption::BIT_WIDTH_MAX_WINDOW, &window),
```

Comment:
`set_option` isn't C.41-compliant; you can't use it here.

You'll need to use an analogue of `FilterCreate::deserialize()` here, and you'll need to write it. In #2799, that function was already present because there had been a previous PR putting it in place. Look at its implementation. All the constructors of individual filters are C.41; `set_option` isn't present.